theme-ui/index.js

```diff
@@ -51,25 +51,18 @@ const aliases = {
 
 const alias = n => aliases[n] || n
 
-const themed = key => theme => css(get(theme, `styles.${key}`))(theme)
+const themed = key => props => css(get(props.theme, `styles.${key}`))(props.theme)
 
-export const Styled = React.forwardRef(({
-  tag = 'div',  // tag is used as a key in theme.styles
-  as,           // as replaces the rendered element type
-  ...props
-}, ref) =>
-  jsx(as || alias(tag), {
-    ...props,
-    ref,
-    css: themed(tag)
-  })
+export const Styled = styled('div')(
+  props => themed(props.tag)(props),
+  css()
```

Comment:
With the changes in #17 – this could just use the `space` utility from core styled-system for margin and padding only